runtime/rasdump/dmpsup.c

```diff
@@ -603,7 +637,7 @@ configureDumpAgents(J9JavaVM *vm)
 						return J9VMDLLMAIN_SILENT_EXIT_VM;
 					}
 					agentOpts[agentNum].kind = kind;
-					agentOpts[agentNum].flags = J9RAS_DUMP_OPT_ARGS_STATIC;
+					agentOpts[agentNum].flags = isMappedToolDump ? J9RAS_DUMP_OPT_ARGS_ALLOC_TOOLMAP : J9RAS_DUMP_OPT_ARGS_STATIC;
```

Comment:
@keithc-ca @pshipton 
I have pushed a new approach to free the allocated memory for `optionString`. I'm not sure if this is the best approach as it introduces a new flag: https://github.com/eclipse/openj9/blob/22fe5f4ff24be7e1bd5a9f32d4d612609f2ccdc9/runtime/rasdump/rasdump_internal.h#L77

The reason for this flag is to mark each mapped tool option such that each can be freed accordingly. The original memory allocation for `optionString` is for `tool:events=systhrow,filter=java/lang/OutOfMemoryError,exec=java -version`: https://github.com/eclipse/openj9/blob/22fe5f4ff24be7e1bd5a9f32d4d612609f2ccdc9/runtime/rasdump/dmpsup.c#L593-L599

However, `optionString` is modified to point to `events=systhrow,filter=java/lang/OutOfMemoryError,exec=java -version` in the existing code to only capture the option section for the tool dump: https://github.com/eclipse/openj9/blob/22fe5f4ff24be7e1bd5a9f32d4d612609f2ccdc9/runtime/rasdump/dmpsup.c#L627-L631

As such, the pointer is incremented by 5 to skip over "tool:". Thus, to free the originally allocated memory, I have created the flag for this case so that we use the address originally used to allocate `optionString`, which is at`agentOpts[i].args - strlen("tool:")`: https://github.com/eclipse/openj9/blob/22fe5f4ff24be7e1bd5a9f32d4d612609f2ccdc9/runtime/rasdump/dmpsup.c#L743-L750

Your feedback would be greatly appreciated!

Reply:
Perhaps I'm missing something: if the string in `agentOpts[i].args` is allocated, why don't we simply put the right value there (after `tool:`) so we don't need a new flag?

Reply:
I think the string in `agentOpts[i].args` is not allocated in the general case (the flag is usually set to `J9RAS_DUMP_OPT_ARGS_STATIC`). 

The reason that the allocation in my implementation is made at `tool:` is because `typeString` is expected to hold all of `tool:events=systhrow,filter=java/lang/OutOfMemoryError,exec=java -version` and `optionString` is only `events=systhrow,filter=java/lang/OutOfMemoryError,exec=java -version`

Reply:
Actually, I think the answer is simple: `optionString` is the value returned by `j9mem_allocate_memory()` which is the same value you must pass to `j9mem_free_memory()`.

You don't *want* to backup by the length of "tool:" and so no new flag is required.